cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/WorkflowHttpHandler.java

```diff
@@ -158,6 +167,9 @@ public void handle(WorkflowClient.Status status) {
                                            if (status.getCode() == WorkflowClient.Status.Code.NOT_FOUND) {
                                              responder.sendStatus(HttpResponseStatus.NOT_FOUND);
                                            } else if (status.getCode() == WorkflowClient.Status.Code.OK) {
+                                             // This uses responder.sendByteArray because status.getResult returns a
+                                             // json string, and responder.sendJson would need deserialization and
```

Comment:
yeah, I saw this unusual usage of ``sendByteArray`` with the ``application/json`` header, so tried to change it to use ``sendJson``. But the usage seems to be to get the response content as a string, and parse it as a json with type ``List<WorkflowActionNode>``. There are a couple of tests that do this. 

``status.getResult`` is a json string, and not an object perhaps because it is populated with the response from the ``/status`` API from the workflow http service. I added a comment so people should not see this as an anomaly and spend time trying to change it to ``responder.sendJson`` in future.